cmd/state/stateless/state.go

```diff
@@ -226,44 +251,35 @@ beginTx:
 			return err
 		}
 
-		skipFirst := len(r.HistoryKey) > 0 // snapshot stores last analyzed key
 		for k, vIsEmpty, err := c.SeekKey(r.HistoryKey); k != nil || err != nil; k, vIsEmpty, err = c.NextKey() {
 			if err != nil {
 				return err
 			}
-			if skipFirst {
-				skipFirst = false
-				continue
+			i++
+			r.HistoryKey = k
+			if r.interrupt(ctx, i, startTime) {
+				return nil
 			}
 
 			copy(addrHash[:], k[:32]) // First 32 bytes is the hash of the address, then timestamp encoding
-			addr := string(addrHash.Bytes())
 			timestamp, _ := dbutils.DecodeTimestamp(k[32:])
+
+			if timestamp > r.StartedWhenBlockNumber { // skip what happened after analysis started
+				continue
+			}
+
 			if timestamp+1 > r.MaxTimestamp {
 				r.MaxTimestamp = timestamp + 1
 			}
 			if vIsEmpty {
 				r.CreationsByBlock[timestamp]++
-				if lt, ok := r.LastTimestamps.Get(addr); ok {
+				if lt, ok := r.lastTimestamps.Get(addrHash.Bytes()); ok {
 					r.CreationsByBlock[lt]--
 				}
 			}
 
-			r.LastTimestamps.Set(addr, timestamp)
-			count++
-			if count%PrintProgressEvery == 0 {
-				fmt.Printf("Processed %dK account records, %s\n", count/1000, time.Since(startTime))
-			}
-			if count%PrintMemStatsEvery == 0 {
-				PrintMemUsage()
-			}
-			if count%SaveSnapshotEvery == 0 {
-				r.HistoryKey = k
-				r.save(ctx)
-			}
-			if count%MaxIterationsPerTx == 0 {
-				r.HistoryKey = k
-				return nil
+			if err := r.lastTimestamps.Put(addrHash.Bytes(), timestamp); err != nil {
```

Comment:
You will need to make a copy of the `addrHash` slice here, because `Bytes()` does not do it. Otherwise, during the commit, you will end up with lots of values overriding the same key (since `addrHash` is reused)

Reply:
Yep, i catched this specific of bolt 5 times already, but still doing same mistake.